b1ddi/resource_ipam_subnet.go

```diff
@@ -340,54 +345,82 @@ func resourceIpamsvcSubnet() *schema.Resource {
 
 func resourceIpamsvcSubnetCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	c := m.(*b1ddiclient.Client)
+	var id, address string
+
+	address = d.Get("address").(string)
+
+	if strings.HasPrefix(address, "ipam/address_block") {
+		params := &address_block.AddressBlockCreateNextAvailableSubnetParams{
+			ID:       address,
+			Cidr:     swag.Int32(int32(d.Get("cidr").(int))),
+			Name:     swag.String(d.Get("name").(string)),
+			Comment:  swag.String(d.Get("comment").(string)),
+			DhcpHost: swag.String(d.Get("dhcp_host").(string)),
+			Context:  ctx,
+		}
 
-	dhcpOptions := make([]*models.IpamsvcOptionItem, 0)
-	for _, o := range d.Get("dhcp_options").([]interface{}) {
-		if o != nil {
-			dhcpOptions = append(dhcpOptions, expandIpamsvcOptionItem(o.(map[string]interface{})))
+		resp, err := c.IPAddressManagementAPI.AddressBlock.AddressBlockCreateNextAvailableSubnet(
+			params,
+			nil,
+		)
```

Comment:
This only creates a subnet with limited fields. How will an user create a next available subnet will some field that is not supported by this call? For e.g. for a resource definition as below - 
```
// Create a subnet dynamically under the Address Block 'tf_example_address_block.'
// Next available subnet under an Address Block example
resource "b1ddi_subnet" "example_tf_subnet_1" {
  name = "tf_subnet_dynamic_nas"
  address = b1ddi_address_block.tf_example_address_block.id
  cidr = 28
  comment = "subnet created through Terraform using Address block ID"
  space = b1ddi_ip_space.space.id
  
  # the following are unsupported in CreateNextAvailableSubnet call
  tags = {
     tag1 = "value1"
  } 
}
```
I believe we should call an Update call after CreateNextAvailableSubnet to update the subnet with the remaining fields.

Reply:
Yes. That's the limitation of the Next Available API. As you mentioned, an Update call has to be made to add additional info to the Subnet. I will update the example with this.

Reply:
Or shall we limit the number of fields supported by this operation?

Reply:
> Or shall we limit the number of fields supported by this operation?

In that case we have to update the schema for those fields by adding `ConflictsWith`